src/WCSimPrimaryGeneratorMessenger.cc

```diff
@@ -82,7 +82,52 @@ WCSimPrimaryGeneratorMessenger::WCSimPrimaryGeneratorMessenger(WCSimPrimaryGener
   injectorWavelengthCmd->SetGuidance("Wavelength of the injector laser in nm");
   injectorWavelengthCmd->SetParameterName("injector_wavelength",true);
   injectorWavelengthCmd->SetDefaultValue(435.);
-    
+   
+
+  // Options for alternative light injector (collimator,diffuser, laser ball)
+  // Set the light injector type, index (placeholders)
+  // retrieves position, direction and profile from DB
+
+  lightInjectorCmd = new G4UIcmdWithAString("/mygen/injectorType",this);
+  lightInjectorCmd->SetGuidance("Select light injector");
+  lightInjectorCmd->SetGuidance("[usage] /mygen/injectorType injector_type");
+  lightInjectorCmd->SetGuidance(" injectorType : collimator, diffuser or laserball");
+  lightInjectorCmd->SetParameterName("injectorType",true);
+  lightInjectorCmd->SetCandidates("collimator diffuser laserball");
+  lightInjectorCmd->SetDefaultValue("collimator");
+
+  lightInjectorIdxCmd = new G4UIcmdWithAString("/mygen/injectorIdx",this);
+  lightInjectorIdxCmd->SetGuidance("Set the ID number of the light injector you want to use");
+  lightInjectorIdxCmd->SetGuidance("[usage] /mygen/injectorIdx index");
+  lightInjectorIdxCmd->SetGuidance("see data/lightInjectors.json for indices for now");
+  lightInjectorIdxCmd->SetGuidance(" injectorIdx : 0,1,2, ... ");
```

Comment:
Not sure what the most susinct solution is, but this made me think you can set multiple injectors to be active by using a comma separated string
Maybe `injectorIdx : INT`?